pkg/store/consul/dsstore/consul_store.go

```diff
@@ -40,18 +46,18 @@ func (e CASError) Error() string {
 	return fmt.Sprintf("Could not check-and-set key %q", string(e))
 }
 
-// Make sure functions declared in the Store interface have the proper contract
-var _ Store = &consulStore{}
-
-func NewConsul(client consulutil.ConsulClient, retries int, logger *logging.Logger) Store {
-	return &consulStore{
+func NewConsul(client consulutil.ConsulClient, retries int, logger *logging.Logger) *store {
```

Comment:
I agree with you it shouldn't be private, i'll fix all these. I do think I want to not return interfaces though. Justification: https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a.

```
Go is at its most powerful when interface definitions are small. In the standard library, most interface definitions are a single method. This allows the greatest reuse because implementing an interfaces is easy to do. When programmers code to preemptive interfaces like Auth above, the interfaces tend to explode in method count, which makes the whole point of an interface (exchangeable implementations) even more difficult to accomplish.
```